docs/DesignDocs/CryptoLibConfiguration.md

```diff
@@ -0,0 +1,115 @@
+!MBEDTLS_ECP_DP_BP256R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+!MBEDTLS_ECP_DP_BP384R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+!MBEDTLS_ECP_DP_BP512R1_ENABLED | N/A | Cannot be disabled on OpenSSL. |
+MBEDTLS_ECP_DP_CURVE25519_ENABLED | N/A | OpenSSL supports curve 25519 by default. |
+MBEDTLS_ECP_DP_CURVE448_ENABLED | N/A | OpenSSL supports curve 448 by default. |
+MBEDTLS_ECP_NIST_OPTIM | N/A | No equivalent option in OpenSSL. |
+!MBEDTLS_ECP_RESTARTABLE | N/A | No equivalent option in OpenSSL. |
+MBEDTLS_ECDSA_DETERMINISTIC | N/A | No equivalent option in OpenSSL. |
+!MBEDTLS_KEY_EXCHANGE_PSK_ENABLED | OPENSSL_NO_PSK | Disable pre-shared keys in enclaves until we have a use case. Basic PSK has no perfect forward secrecy, not recommended for future use. Disable PSK on OpenSSL with the no-psk configuration. |
+!MBEDTLS_KEY_EXCHANGE_DHE_PSK_ENABLED | OPENSSL_NO_PSK | Disable pre-shared keys in enclaves until we have a use case. Disable PSK on OpenSSL with the no-psk configuration. |
+!MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED | OPENSSL_NO_PSK | Disable pre-shared keys in enclaves until we have a use case. Disable PSK on OpenSSL with the no-psk configuration. |
+!MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED | OPENSSL_NO_PSK | Disable pre-shared keys in enclaves until we have a use case. RSA PSK has no perfect forward secrecy, not recommended for future use. Disable PSK on OpenSSL with the no-psk configuration. |
+!MBEDTLS_KEY_EXCHANGE_RSA_ENABLED | N/A | Deprecated in v0.7, developers should consider ECDHE key exchange instead for forward secrecy. No equivalent option in OpenSSL. The following ciphersuites are supported by OpenSSL: `RSA_WITH_AES_256_GCM_SHA384`, `RSA_WITH_AES_256_CBC_SHA`, `RSA_WITH_CAMELLIA_256_CBC_SHA256`, `RSA_WITH_CAMELLIA_256_CBC_SHA`, `RSA_WITH_AES_128_GCM_SHA256`, `RSA_WITH_AES_128_CBC_SHA`, `RSA_WITH_CAMELLIA_128_CBC_SHA256`, and `RSA_WITH_CAMELLIA_128_CBC_SHA`. |
+!MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED | N/A | Not supported in favor of ECDHE for performance. No equivalent option in OpenSSL. The following ciphersuites are supported by OpenSSL: `DHE_RSA_WITH_AES_256_GCM_SHA384`, `DHE_RSA_WITH_CAMELLIA_256_CBC_SHA256`, `DHE_RSA_WITH_CAMELLIA_256_CBC_SHA`, `DHE_RSA_WITH_AES_128_GCM_SHA256`, `DHE_RSA_WITH_CAMELLIA_128_CBC_SHA256`, and `DHE_RSA_WITH_CAMELLIA_128_CBC_SHA`. |
+MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED | N/A | OpenSSL supports ECDHE-RSA based ciphersuites by default. No equivalent option in OpenSSL. |
+MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED | N/A | OpenSSL supports ECDHE-ECDSA based ciphersuites by default. No equivalent option in OpenSSL. |
+!MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED | N/A | Drop uncommon protocol to minimize TCB. No equivalent option in OpenSSL. |
+!MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED | N/A | Drop uncommon protocol to minimize TCB. No equivalent option in OpenSSL. |
+!MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED | N/A | No equivalent option in OpenSSL. |
```

Comment:
This should also comment that "OpenSSL does not support JPAKE (removed in v1.1.0)" which is a clearer explanation used for MBEDTLS_ECJPAKE_C

Reply:
Updated. Thanks for the information!